From da823e27ec906cc4c6bed4ace7913fd37bec72cc Mon Sep 17 00:00:00 2001
From: xe5700 <9338143+xe5700@users.noreply.github.com>
Date: Thu, 19 May 2022 18:56:34 +0800
Subject: [PATCH 1/1] Support for old version python 3

---
 htserver.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kvmd/htserver.py b/kvmd/htserver.py
index c24837d..4a3c306 100644
--- a/kvmd/htserver.py
+++ b/kvmd/htserver.py
@@ -297,7 +297,7 @@ class HttpServer:
             shutdown_timeout=1,
             access_log_format=access_log_format,
             print=self.__run_app_print,
-            loop=asyncio.get_event_loop(),
+            #loop=asyncio.get_event_loop(),
         )
 
     # =====
-- 
2.34.1.windows.1

