From 368eaa19ef1cc187c8012c00b95ad97f260d61c3 Mon Sep 17 00:00:00 2001
From: xe5700 <9338143+xe5700@users.noreply.github.com>
Date: Sun, 23 Oct 2022 11:10:24 +0800
Subject: [PATCH] Disable-GPIO-For-TV-Box

---
 kvmd/apps/kvmd/ugpio.py | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/kvmd/apps/kvmd/ugpio.py b/kvmd/apps/kvmd/ugpio.py
index a8fc9224..8bf17fff 100644
--- a/kvmd/apps/kvmd/ugpio.py
+++ b/kvmd/apps/kvmd/ugpio.py
@@ -281,16 +281,10 @@ class UserGpio:
             await self.__notifier.wait()
 
     def sysprep(self) -> None:
-        get_logger().info("Preparing User-GPIO drivers ...")
-        for (_, driver) in tools.sorted_kvs(self.__drivers):
-            driver.prepare()
+        pass
 
     async def systask(self) -> None:
-        get_logger(0).info("Running User-GPIO drivers ...")
-        await asyncio.gather(*[
-            driver.run()
-            for (_, driver) in tools.sorted_kvs(self.__drivers)
-        ])
+        await asyncio.Event().wait()
 
     async def cleanup(self) -> None:
         for driver in self.__drivers.values():
-- 
2.34.1.windows.1

